### PR TITLE
Sbt 73 Admin Defined Object dump + dump templates

### PR DESCRIPTION
**Pull request by quantori-cherednichenko**

### CODE REVIEW

File: src/signals_notebook/entities/admin_defined_object.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    ado_name: AdoTypeName = Field(alias='adoName')
+
+    class Config:
+        validate_assignment = True
+        allow_population_by_field_name = True
+
+
+class AdminDefinedObject(Container):
+    class TypeName(str, Enum):
+        NEW_OBJECT = 'New System Object (SK)'
```
Comment (quantori-pokidovea) [inline src/signals_notebook/entities/admin_defined_object.py:62]: SK - Sergey Kuznetsov

File: src/signals_notebook/entities/admin_defined_object.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    type: Literal[EntityType.ADO] = Field(allow_mutation=False)
+    ado: AdoType[TypeName]
+    _template_name: ClassVar = 'ado.html'
+
+    class Config:
+        keep_untouched = (cached_property,)
+
+    def __init__(self, *args, **kwargs):
+        # self.ado_type_name =
```
Comment (quantori-pokidovea) [inline src/signals_notebook/entities/admin_defined_object.py:74]: remove?

**Review by quantori-pokidovea — APPROVED** [review]

File: src/signals_notebook/entities/admin_defined_object.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    @classmethod
+    def create(
+        cls,
+        *,
+        name: str,
+        ado_type_name: str,
+        description: Optional[str] = None,
+        template: Optional['AdminDefinedObject'] = None,
+        notebook: Optional[Notebook] = None,
+        digest: str = None,
```
Comment (quantori-shishkin) [inline src/signals_notebook/entities/admin_defined_object.py:78]:
```suggestion
        digest: Optional[str]= None,
```

File: src/signals_notebook/entities/admin_defined_object.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                meta=_Meta(ado_type_name=ado_type_name),
+                attributes=_Attributes(
+                    name=name,
+                    description=description,
+                ),
+                relationships=relationships,
+            )
+        )
+
+        log.debug('Creating Notebook for: %s', cls.__name__)
```
Comment (quantori-shishkin) [inline src/signals_notebook/entities/admin_defined_object.py:113]: Creating ....

File: src/signals_notebook/entities/admin_defined_object.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+        }
+        fs_handler.write(
+            fs_handler.join_path(base_path, self.eid, 'metadata.json'),
+            json.dumps(metadata),
+        )
+        for child in self.get_children():
+            child.dump(fs_handler.join_path(base_path, self.eid), fs_handler)
+
+    @classmethod
+    def load(cls, path: str, fs_handler: FSHandler, notebook: Notebook) -> None:
```
Comment (quantori-shishkin) [inline src/signals_notebook/entities/admin_defined_object.py:151]: docstring

File: src/signals_notebook/entities/admin_defined_object.py [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            'edited_at': self.edited_at,
+            'children': self.get_children(),
+        }
+
+        template = env.get_template(self._template_name)
+        log.info('Html template for %s:%s has been rendered.', self.__class__.__name__, self.eid)
+
+        return template.render(data=data)
+
+    def dump(self, base_path: str, fs_handler: FSHandler) -> None:
```
Comment (quantori-shishkin) [inline src/signals_notebook/entities/admin_defined_object.py:138]: docstring

**Review by quantori-shishkin — APPROVED** [review]

**Pull request merged**